### PR TITLE
Rocksdb replication

### PR DESCRIPTION
**Pull request by michaelpanpliops**

### CODE REVIEW

File: pliops/producer.cc [hunk 1]
```diff
@@ -96,7 +96,9 @@ void Producer::ReaderThread(uint32_t iterator_parallelism_factor, uint32_t threa
 
     if (!EnqueueTimed(message_queue_, {std::string(key.data(), key.size()), std::string(value.data(), value.size())}, kill_, ops_timeout_msec_)) {
       // We must release the iterator here, otherwise DB::Close will crash
-      status = iterator->Close();
+#ifdef XDPROCKS
+      iterator->Close();
+#endif
```
Comment (tamarwo16) [inline pliops/producer.cc:101]: why remove the status and not add it to the error message?

File: pliops/producer.cc [hunk 2]
```diff
@@ -113,7 +115,9 @@ void Producer::ReaderThread(uint32_t iterator_parallelism_factor, uint32_t threa
   logger->Log(Severity::INFO, FormatString("Reader thread #%d ended. Performed %lld operations.\n", thread_id, total_number_of_operations));
 
   // Release the iterator
+#ifdef XDPROCKS
   status = iterator->Close();
+#endif
```
Comment (tamarwo16) [inline pliops/producer.cc:120]: why do only Xdprocks need to call iterator->Close()?

File: pliops/producer.cc (hunk 1 above)
Comment (michaelpanpliops) [reply pliops/producer.cc:101]: this API only exists in XdpRocks

File: pliops/producer.cc (hunk 2 above)
Comment (michaelpanpliops) [reply pliops/producer.cc:120]: this API only exists in XdpRocks

**Pull request merged**